### PR TITLE
Llama tutorial fixes

### PR DESCRIPTION
**Pull request by pggPL**
I fixed a few bugs in Llama tutorial:

1. Weight loading. 

There was a problem with loading official Llama weights from some sources, caused by the bug in the code. Weights downloaded from official meta website and converted to the HF format were not working, but weights downloaded directly from the HF weren't causing any problems.

Models weights were contained in a few files(3 or 4, depending of the way they were downloaded). The problem was caused by the assumption in the code, that every layer's weights need to be in one file. It was the case with the HF weights, but not with the converted weights from Meta website.

I fixed this by changing ifs in the [te_llama.py](https://github.com/NVIDIA/TransformerEngine/compare/main...pggPL:TransformerEngine:Llama-tutorial-modifications?expand=1#diff-76ffe7b895af37a2ef5bb5365e2058eda360d946b2e3bde067784f07cfac677b) file. There are basically 2 dictionaries there - `hf_state_dict` and `te_state_dict` - and we want to copy some params from the first to the second - maybe with some name change. It's easy to see that we were checking wrong dictionary in the if. After change it started to work.

2. Bug caused by HuggingFace commit.

[Commit](https://github.com/huggingface/transformers/commit/ff841900e45763114d2417fb24ce29d950c6c956) on HF made tutorial crash. The reason was that HF code refered to the name of the parameter, which was changed when model was migrated from the LlamaDecoderLayer to TEDecoderLayer. The line `config._attn_implementation = "flash_attention_2"` solves this problem, because HF code does not pass through if, which access problematic parameter/

3. Bug with group query attention.

Even though Llama uses groupped query attention, numbers of kv heads were not passes to the TEDecoderLayer. The reason it wouldn't crash is that in fact this model of Llama uses num_gqa_groups=num_heads, which is the default value.

### CODE REVIEW

**Comment by ksivaman** [comment]
@pggPL Could you sign-off your commit according the guide [here](https://github.com/NVIDIA/TransformerEngine/blob/c1a68f6cb40a73d7e7bf77aa73c2820ffaf47281/CONTRIBUTING.rst#sign-your-work) Thanks!.

File: docs/examples/te_llama/te_llama.py [hunk 1]
```diff
@@ -121,12 +121,10 @@ def from_pretrained_local(cls, pretrained_model_name_or_path, *args, config, **k
...
             resolved_archive_file = [resolved_archive_file]
 
-        error_msgs = []
         for shard_file in resolved_archive_file:
             state_dict = load_state_dict(shard_file)
-            replaced_layers = replace_params(state_dict, vanilla_model.state_dict())
-
-            error_msgs += _load_state_dict_into_model(vanilla_model, state_dict, start_prefix="")
+            replace_params(state_dict, vanilla_model.state_dict())
+            _load_state_dict_into_model(vanilla_model, state_dict, start_prefix="")
```
Comment (sudhakarsingh27) [inline docs/examples/te_llama/te_llama.py:127]: Now I feel that this line is maybe not needed. Could you verify, @pggPL ?

File: docs/examples/te_llama/te_llama.py [hunk 2]
```diff
@@ -142,32 +140,31 @@ def replace_params(hf_state_dict, te_state_dict):
         m = re.match(layer_prefix_pat, param_key)
         if m is not None:
             all_layer_prefixes.add(m.group())
-
+
```
Comment (ptrendx) [inline docs/examples/te_llama/te_llama.py:143]: nit: whitespace

File: docs/examples/te_llama/te_llama.py (hunk 1 above)
Comment (pggPL) [reply docs/examples/te_llama/te_llama.py:127]: Seems that it's necessary - results without it are different than with it. I will investigate it tomorrow.

Comment (pggPL) [reply docs/examples/te_llama/te_llama.py:127]: We do not load all params in replace_params - for example embedding, thus this line is necessary.

File: docs/examples/te_llama/utils.py [hunk 3]
```diff
@@ -91,6 +91,7 @@ def init_te_llama_model(hyperparams):
     # Init the model
     from te_llama import TELlamaForCausalLM
     config = AutoConfig.from_pretrained(hyperparams.model_name)
+    config._attn_implementation = "flash_attention_2"
```
[the remark this answers is not in the document]
Comment (pggPL) [reply docs/examples/te_llama/utils.py:94]: I've added.

File: docs/examples/te_llama/te_llama.py [hunk 4]
```diff
@@ -142,32 +140,38 @@ def replace_params(hf_state_dict, te_state_dict):
         m = re.match(layer_prefix_pat, param_key)
         if m is not None:
             all_layer_prefixes.add(m.group())
+    
+    GATE_PROJ_SIZE = 11008
```
Comment (sudhakarsingh27) [inline docs/examples/te_llama/te_llama.py:144]: Can we use `config` object to get this value? We could pass `config` object to `replace_params` method.

**Review by sudhakarsingh27 — APPROVED** [review]
Lgtm

**Pull request merged**